independent-projects/bootstrap/maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -274,16 +283,50 @@ private void transformLegacyToNew(final Path output, ObjectNode extObject, Objec
             metadata.set("short-name", extObject.get("shortName"));
             extObject.remove("shortName");
         }
-
+        // Add quarkus core version used to build this extension
+        String quarkusCoreVersion = resolveQuarkusCoreVersion();
+        if (quarkusCoreVersion != null) {
+            metadata.put("quarkus-core", quarkusCoreVersion);
+        }
         extObject.set("metadata", metadata);
+    }
 
-        //   updateSourceFiles(output, extObject, mapper);
+    private String resolveQuarkusCoreVersion() throws MojoExecutionException {
+        MavenArtifactResolver mvn = getMavenResolver();
+        DefaultArtifact artifact = new DefaultArtifact(project.getGroupId(),
+                project.getArtifactId(),
+                project.getPackaging(),
+                project.getVersion());
+        final CollectResult collectResult;
+        try {
+            collectResult = mvn.collectDependencies(artifact, Collections.emptyList());
+        } catch (AppModelResolverException e) {
+            return null;
+        }
+        final AtomicReference<String> version = new AtomicReference<>();
+        final TreeDependencyVisitor visitor = new TreeDependencyVisitor(new DependencyVisitor() {
+            @Override
+            public boolean visitEnter(DependencyNode node) {
+                return version.get() == null;
+            }
 
+            @Override
+            public boolean visitLeave(DependencyNode node) {
+                final Dependency dep = node.getDependency();
+                if (dep != null
+                        && dep.getArtifact().getArtifactId().equals("quarkus-core")
+                        && dep.getArtifact().getGroupId().equals("io.quarkus")) {
```

Comment:
There are extensions that don't depend on `quarkus-core` and it's not necessarily a mistake. However every deployment artifact has `quarkus-core-deployment` among its deps. So, that's what you should be looking for instead.

Reply:
The problem is that the `quarkus-extension.yaml` is in the runtime module, and the deployment module is a dependency of of this module

Reply:
Is there a reason to keep it in the runtime artifact?

Reply:
Dunno, it has always been like this AFAIK.